src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoSink.java

```diff
@@ -0,0 +1,328 @@
+            
+            // this is used for storing the attribute's names (sufixed with "-md") and metadata when dealing with a per
+            // column attributes persistence; in that case the persistence is not done attribute per attribute, but
+            // persisting all of them at the same time
+            HashMap<String, String> mds = new HashMap<String, String>();
+
+            for (NotifyContextRequest.ContextAttribute contextAttribute : contextAttributes) {
+                String attrName = contextAttribute.getName();
+                String attrType = contextAttribute.getType();
+                String attrValue = contextAttribute.getContextValue(false);
+                String attrMetadata = contextAttribute.getContextMetadata();
+                LOGGER.debug("[" + this.getName() + "] Processing context attribute (name=" + attrName + ", type="
+                        + attrType + ")");
+                
+                // create the collection at this stage, if the data model is collection-per-attribute
+                if (dataModel == DataModel.COLLECTIONPERATTRIBUTE) {
+                    collectionName = buildCollectionName(fiwareServicePath, destinations[i], attrName);
+                    backend.createCollection(dbName, collectionName);
+                } // if
+
+                LOGGER.info("[" + this.getName() + "] Persisting data at OrionMongoSink. Database: " + dbName
+                        + ", Collection: " + collectionName + ", Data: " + recvTimeTs / 1000 + "," + recvTime + ","
+                        + entityId + "," + entityType + "," + attrName + "," + entityType + "," + attrValue + ","
+                        + attrMetadata);
+                backend.insertContextDataRaw(dbName, collectionName, recvTimeTs / 1000, recvTime,
+                        entityId, entityType, attrName, attrType, attrValue, attrMetadata);
+            } // for
+        } // for
+    } // persist
+ 
+    /**
+     * Gets the data model given its string representation.
+     * @param dataModelStr
+     * @return
+     */
+    protected DataModel getDataModel(String dataModelStr) {
+        if (dataModelStr.equals("collection-per-service-path")) {
+            return DataModel.COLLECTIONPERSERVICEPATH;
+        } else if (dataModelStr.equals("collection-per-entity")) {
+            return DataModel.COLLECTIONPERENTITY;
+        } else if (dataModelStr.equals("collection-per-attribute")) {
+            return DataModel.COLLECTIONPERATTRIBUTE;
+        } else {
+            return null;
+        } // if else if
+    } // getDataModel
+
+    /**
+     * Builds a database name given a fiwareService. It throws an exception if the naming conventions are violated.
+     * @param fiwareService
+     * @return
+     * @throws Exception
+     */
+    private String buildDbName(String fiwareService) throws Exception {
+        String dbName = dbPrefix + fiwareService;
```

Comment:
MongoDB imposes some limitations on the characters a database may use. See http://docs.mongodb.org/manual/reference/limits/#naming-restrictions Consequently, I suggest substituting those characters for some special character :) In fact, I would appreciate if you tell me which that character is or how to deal with this to implement the same behaviour in the STH component :) In fact, we can use this comment to discuss about it :)